### ISSUE TITLE
Use a consistent color for all controls and inputs labels

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#66454]
### Description

It appears there's some inconsistency with the color used for various labels in the editor. I'm not sure what is the value of using different shades of gray, I'd tend to think all labels should use the same color. 

Also, for accessibility, it would be good to not just aim to be compliant with the _minimum_ required color contrast ratio. It would be good to use a gray that is darker than `#757575` (`$gray-700`). On a white background, `#757575` produces a contrast ratio of 4.61:1 that is just a little above the minimum required of 4.5:1. I'd argue that `$gray-700` may be used for non-essential information e.g. descriptions but labels should be more readable. 

A few examples taken from the Styles > Background UI. There may be more cases, these are just examples.

![Image](https://github.com/user-attachments/assets/2076acd3-6f92-45c9-a3a2-b5cc30ed892a)

1. `$gray-900` - `#1e1e1e` - I would like this to be the standard for all labels as it's the shade of gray that provides better readability.
2. `#3c434a` - This is inherited from the body, it comes from the wp-admin `common.css`. I'd say this is a bug because the editor shouldn't rely on colors provided by external stylesheets.
3. `$gray-700` - `#757575` - This is unnecessarily too light.

## The ToggleGroupControl case

It could be argued that the non-selected options in the ToggleGroupControl use a lighter gray because they are 'inactive'. To me, that's not correct. These are _enabled_ controls that can be selected. Rather than being 'inactive', they are just not-selected. Under the hood, the ToggleGroupControl is the equivalent of a group of radio buttons. The labels of a group of radio buttons use the same color. There's no need to communicate the 'selected' state by the means of color because the selection state is inherently communicated by the visuals of the selected radio button. In the same way, there's no need to use a lighter gray in the ToggleGroupControl because the selected option is visually distinguished by a a very noticeable inversion of colors:

![Image](https://github.com/user-attachments/assets/07b51c3d-7040-4646-8da0-1d735b228177)

On top of that, for accessibility reasons, there's no reason to make the not-selected options _less readable_. I would expect these to use the normal gray used for other labels.

Also, when there is no option selected, I'd argue that they all look 'disabled' because of the lighter gray. Screenshot:

![Image](https://github.com/user-attachments/assets/b7725bce-e785-4fc7-a779-706fd0fac063)

![Image](https://github.com/user-attachments/assets/9f83960b-c3dd-49d5-9cdd-bf4db606d367)

For consistency and better accessibility, I'd rather consider to standardize the color for all labels.

### Step-by-step reproduction instructions

- Go through the examples above.
- Observe the labels use different shades of gray.
- Optionally check other labels of other controls.

### Please confirm that you have searched existing issues in the repo.

- [x] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [x] Yes

### Please confirm which theme type you used for testing.

- [ ] Block
- [ ] Classic
- [ ] Hybrid (e.g. classic with theme.json)
- [ ] Not sure

**Comment by t-hamano** [comment]
I think this issue potentially contains a broader concern. This means that there are components whose label color or text changes depending on where they are placed.

Let's use the `ToggleGroupControl` as an example.

The label of this component does not have a color. Therefore, in the background image popover, its color is `#3c434a`, as it inherits the default text color from the WP-Admin style:

![Image](https://github.com/user-attachments/assets/e159d7c9-2890-4945-95ad-153a42001170)

On the other hand, in the block sidebar, the label color will be `#1e1e1e` (`$gray-900`) due to the definition [here](https://github.com/WordPress/gutenberg/blob/trunk/packages/interface/src/components/complementary-area/style.scss#L3):

![Image](https://github.com/user-attachments/assets/1eb820c5-d195-4e57-8377-92e362d258ea)

As a result, the label color changes depending on where this component is placed.

On the other hand, for example, the label of the `InputControl` component is based on the `Text` component and has [a defined text color](https://github.com/WordPress/gutenberg/blob/961000076e74a6eea419d79ff8e16ae54a73463b/packages/components/src/text/styles.ts#L12), so they will not be overridden by WP-Admin styles.

Maybe we need to review the labels on all components and apply a uniform color. Also, not only the labels but also the text of the `CheckboxControl`, `RadioControl`, and `ToggleControl` components do not have color, so we might need to consider them at the same time.

cc @WordPress/gutenberg-components

**Comment by afercia** [comment]
> Also, not only the labels but also the text of the `CheckboxControl`, `RadioControl`, and `ToggleControl` components do not have color,

Good catch. And yes, I'd agree the case where color isn't set and thus inherits from the WP admin context should be prevented.

**Comment by valerogarte** [comment]
I agree that unifying label colors is essential for enhancing accessibility and visual consistency. It's also crucial to review all components to ensure they function as expected. However, I wonder if it would be more effective to implement this change at a global level to ensure it applies universally. This is my second PR in WordPress, and I'm open to your suggestions.

**Comment by ciampo** [comment]
I agree that this warrants a broader conversation. Should every component define its text color, or should it inherit it from its context / CSS cascade?

We would probably prefer the former, also given that we have plans to implement theming (which would easily allow consumers to tweak text color anyway). But IMO the most important thing would be to ensure consistency across the package.

**Comment by valerogarte** [comment]
To move forward with this issue and unblock the task, how should we proceed? Where and how is this topic being discussed?

**Comment by mirka** [comment]
I suggest we start with an audit of our main "control" components to see which ones need fixing.

These are listed in Storybook under [Components ▸ Selection & Input ▸ Common](https://wordpress.github.io/gutenberg/?path=/docs/components-basecontrol--docs) and [Components (Experimental) ▸ Selection & Input](https://wordpress.github.io/gutenberg/?path=/docs/components-experimental-inputcontrol--docs).

A quick way to check if labels have explicit colors set is to use the experimental theme switcher tool. If the label is not visible in dark mode, it likely doesn't have an explicit color set. (Or does not a have a theme-ready color variable set, which we should take this opportunity to fix!)

https://github.com/user-attachments/assets/638bb6f6-6f7f-40e8-9f59-402d5e2f9a7c

**Comment by t-hamano** [comment]
I looked into components whose labels become invisible in dark mode, including components with the `__experimental` prefix. Some components are combinations of several components, so the number of components we have to deal with should actually be less than it actually is.

<details><summary>AnglePickerControl</summary>

![Image](https://github.com/user-attachments/assets/30c4c08e-d7cc-45a2-a70a-5a8831b413bd)

</details>

 <details><summary>CustomGradientPicker</summary>

![Image](https://github.com/user-attachments/assets/e6a7e082-236a-43aa-8159-3ec8aaffe936)

</details>

<details><summary>FocalPointPicker</summary>

![Image](https://github.com/user-attachments/assets/12443468-07eb-4416-aa13-67e75c58ca62)

</details> 

 <details><summary>GradientPicker</summary>

![Image](https://github.com/user-attachments/assets/9dfeb430-11e7-4f28-be85-c8d03c06691a)

</details>

<details><summary>InputControl</summary>

![Image](https://github.com/user-attachments/assets/9ae5d34c-698e-40f2-aa61-9fac426a1f55)

</details> 

 <details><summary>NumberControl</summary>

![Image](https://github.com/user-attachments/assets/077fcb9a-0715-4e9a-aa48-c6fd85ef4b55)

</details>

<details><summary>PaletteEdit</summary>

![Image](https://github.com/user-attachments/assets/bc6cfb18-b517-4885-b152-8d6d37414ad4)

</details> 

 <details><summary>QueryControl</summary>

![Image](https://github.com/user-attachments/assets/b06ff58d-0b82-4415-8a7c-0d5cd7f17fb7)

</details>

<details><summary>SelectControl</summary>

![Image](https://github.com/user-attachments/assets/45dd6235-0e25-4674-925d-eff24087edf6)

</details> 

 <details><summary>TreeSelect</summary>

![Image](https://github.com/user-attachments/assets/93864df4-9881-401e-a1a3-326000f35a0d)

</details>

<details><summary>UnitControl</summary>

![Image](https://github.com/user-attachments/assets/8ad2d15a-ce5e-4b9b-83b3-80723cebc083)

</details>

**Comment by ciampo** [comment]
Hopefully this should be as easy as changing the text color to `--wp-components-color-foreground` for the time being. 

In the future, it would be nice if we could have all (most) controls use `BaseControl` for better consistency (related: https://github.com/WordPress/gutenberg/issues/60836)

**Comment by im3dabasia** [comment]
> I looked into components whose labels become invisible in dark mode.

Hey @t-hamano , Thanks for pointing this out. Can I raise a PR for this fix? I can create a new issue or link this [comment](https://github.com/WordPress/gutenberg/issues/66454#issuecomment-2556154466) in the PR description. 

Looking forward to your thoughts.

**Comment by t-hamano** [comment]
@im3dabasia Thanks for taking an interest in this issue.

> Can I raise a PR for this fix? I can create a new issue or link this https://github.com/WordPress/gutenberg/issues/66454#issuecomment-2556154466 in the PR description.

The problem I raised is part of this issue, so I think it would be fine to just submit a PR instead of submitting a new issue.

**Comment by tyxla** [comment]
#68349 has just landed, thanks @im3dabasia for the help!

**Comment by t-hamano** [comment]
Update: 

> I looked into components whose labels become invisible in dark mode

This issue has been resolved by #68349.

In addition, I discovered that the text of the Notice component becomes invisible in dark mode, so I would like to address this issue.

https://wordpress.github.io/gutenberg/?path=/story/components-notice--default&globals=componentsTheme:darkBg

![Image](https://github.com/user-attachments/assets/41890a7c-ac1f-4989-93c7-015e83041928)

**Comment by SainathPoojary** [comment]
Hey @t-hamano,

I noticed that components using ToolbarGroup with an SVG appear white in dark mode. This happens because the SVG inherits currentColor, while the parent ComponentButton uses the foreground color, which is also white making the SVG invisible due button background color being white. This issue seems to occur in multiple places wherever ToolbarGroup is used with an SVG.

<img width="521" alt="Image" src="https://github.com/user-attachments/assets/2ec5b90c-55e5-4990-ab26-f85ec571709a" />

Some instance:

<img width="447" alt="Image" src="https://github.com/user-attachments/assets/66632edc-b728-43af-8a50-9ec7e6288bc9" />

<img width="408" alt="Image" src="https://github.com/user-attachments/assets/1ac16516-5b08-414d-9689-575b97ad500f" />


Would this be something we need to address? If so, I’d be happy to work on it.

**Comment by ciampo** [comment]
> I noticed that components using ToolbarGroup with an SVG appear white in dark mode. This happens because the SVG inherits currentColor, while the parent ComponentButton uses the foreground color, which is also white making the SVG invisible due button background color being white. This issue seems to occur in multiple places wherever ToolbarGroup is used with an SVG.

The white `fill` for SVG icons seems correct to me, what seems to be missing is a dark background color when in dark mode?

**Comment by SainathPoojary** [comment]
Hey @ciampo,
I’m not sure, but wouldn’t this be the expected behavior in dark mode when the button’s background color is white?

<img width="406" alt="Image" src="https://github.com/user-attachments/assets/19d8c891-6bc0-4af7-9316-37940c956ca3" />


<img width="391" alt="Image" src="https://github.com/user-attachments/assets/a1355ff5-0b21-45e4-8dfa-b33c32f8eb55" />

**Comment by ciampo** [comment]
> wouldn’t this be the expected behavior in dark mode when the button’s background color is white?

Yes, my point is a different one — is the toolbar supposed to have a white background and a dark foreground color when in dark mode? Should it be the other way around? I don't have the latest design specs at handy, but I just wanted to point that out in case it's relevant.

**Comment by t-hamano** [comment]
> I noticed that components using ToolbarGroup with an SVG appear white in dark mode

In Storybook, the easiest way to see this is to look at [the Toolbar component](https://wordpress.github.io/gutenberg/?path=/docs/components-toolbar--docs&globals=componentsTheme:darkBg):

![Image](https://github.com/user-attachments/assets/769dea65-4b61-4eda-b205-b501e6d99901)

The outcome I'm hoping for is either.

Same as when in non-dark mode:

![Image](https://github.com/user-attachments/assets/d7201660-11a1-4582-a981-f92df3978ff4)


Invert colors:

![Image](https://github.com/user-attachments/assets/4937d5af-a78b-43ab-ba1e-4108fb83d228)

**Comment by SainathPoojary** [comment]
Considering the discussion, I’ve raised [this PR](https://github.com/WordPress/gutenberg/pull/69278) to address the issue. Here’s how it looks now in both dark and default modes:
### Dark mode:

<img width="400" alt="Image" src="https://github.com/user-attachments/assets/441c46b6-f4b2-475f-a310-206af695b7c0" />
<br>
<img width="300" alt="Image" src="https://github.com/user-attachments/assets/3e0531a4-cef9-468e-807f-085ad67ef1e5" />
<br>
<img width="300" alt="Image" src="https://github.com/user-attachments/assets/83424ea5-6586-44d6-b420-eeb4882295ac" />

### Default mode:

<img width="400" alt="Image" src="https://github.com/user-attachments/assets/bb514919-0df4-4509-af7c-d5c0cf706aba" />
<br>
<img width="300" alt="Image" src="https://github.com/user-attachments/assets/0327bf5f-e908-4c92-bb8e-fa39d2349402" />
<br>
<img width="300" alt="Image" src="https://github.com/user-attachments/assets/739f26cb-a312-47bb-944c-aadbfa205d36" />


Please have a look whenever you get a moment.

**Comment by t-hamano** [comment]
Note that the issue regarding the theming support has been extracted to #69646.

**Comment by karmatosed** [comment]
This has a design it's more about applying it so let's remove that label for now. If I am wrong though, we can always add that back in.

**Comment by mirka** [comment]
Label colors will be explicitly set in #83318.

I guess the remaining decision points for this issue are just `ToggleGroupControl` now?

**Comment by ciampo** [comment]
This could be an opportunity for reviewing `ToggleGroupControl`'s design holistically as we're soon going to add its counterpart to `@wordpress/ui` (cc @WordPress/gutenberg-design )
